### PR TITLE
add rake setup

### PR DESCRIPTION
**Pull request by lokeshh**
Addresses #180

### CODE REVIEW

**Comment by lokeshh** [comment]
`rake spec`, `rake spec run all`, `rake spec setup` are working.

What about `rubocop`? What should be the task for that and should it run within any of the above tasks?

**Comment by v0dro** [comment]
Hmmm you can the rubocop task within `rake spec run all`. The rubocop output should be present in the test summary.

A separate task for that can be called `rake rubocop`.

File: Rakefile [hunk 1]
```diff
@@ -1,26 +1,51 @@
...
 
-task all: [:cop, :run_all]
+task :spec do
+  case ARGV[1]
+  when 'setup'
+    spec_setup
+  when 'run'
+    spec_run_all if ARGV[2] == 'all'
+  when nil
+    run 'rspec spec'
```
Comment (v0dro) [inline Rakefile]: Should be `bundle exec`....

**Comment by lokeshh** [comment]
I've made all the changes like you suggested. Just one small deviation. Instead of `rake spec run -v 2.3.0`, it works with `rake spec run ruby-2.3.0`.
```ruby
ubuntu@ip-172-31-60-45:~/daru$ rake spec run -v 2.3
rake aborted!
ArgumentError: Now a valid ruby version. Valid ruby versions are ["ruby-2.0.0-p643", "ruby-2.1.5", "ruby-2.2.1", "ruby-2.3.0"]
/home/ubuntu/daru/Rakefile:21:in `block in <top (required)>'
...
ubuntu@ip-172-31-60-45:~/daru$ rake spec run ruby-2.3.0
$rvm_path/wrappers/ruby-2.3.0/rake spec
bundle exec rspec spec
...
```

**Comment by v0dro** [comment]
It's fine. You're using the RVM terminology anyway.

Why is the build failing?

**Comment by lokeshh** [comment]
@v0dro This is ready to be merged.

**Pull request merged**